pkg/ipamd/ipamd.go

```diff
@@ -187,6 +188,38 @@ type IPAMContext struct {
 	terminating            int32 // Flag to warn that the pod is about to shut down.
 }
 
+// UnmanagedENISet keeps a set of ENI IDs for ENIs tagged with "node.k8s.amazonaws.com/no_manage"
+type UnmanagedENISet struct {
+	set  map[string]bool
+	lock sync.RWMutex
+}
+
+func (u *UnmanagedENISet) isUnmanaged(eniID string) bool {
+	val, ok := u.set[eniID]
+	return ok && val == true
+}
+
+func (c *IPAMContext) addUnmanagedENIs(tagMap map[string]awsutils.TagMap) {
+	if len(tagMap) == 0 {
+		return
+	}
+	c.unmanagedENIs.lock.Lock()
+	defer c.unmanagedENIs.lock.Unlock()
+	if len(c.unmanagedENIs.set) == 0 {
+		c.unmanagedENIs.set = make(map[string]bool)
+	}
```

Comment:
and now you can remove the above since you've hidden the implementation details of the lock and map init in the `*UnmanagedENISet.add()` method.